code/extensions/GridFieldBetterButtonsItemRequest.php

```diff
@@ -548,11 +548,35 @@ public function recordIsPublished() {
 
 	}
 
+	/**
+	 * Determines if the current record is deleted from stage
+	 * @returns boolean
+	 */
+	public function recordIsDeletedFromStage() {
+
+		// for SiteTree records
+		if ( $this->owner->hasMethod('getIsDeletedFromStage') ) {
+			return $this->IsDeletedFromStage;
+		}
+
+		if(!$this->owner->record->checkVersioned()) return false;
+		if (!$this->owner->record->isInDB()) {
+			return true;
+		}
+
+		$table = $this->owner->record->class;
```

Comment:
Should this be called $class - minor but then it matches what you are retrieving from the record